File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    map<string, int> m;
    
    for (auto p : participant){
        m[p]++;
    }
    
    for (auto c : completion){
        m[c]--;
    }
    
    for (auto pair : m){
        if (pair.second > 0) return pair.first;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    const participantMap = new Map();
    
    for (const p of participant) {
        participantMap.set(p, (participantMap.get(p) || 0) + 1);
    }
    
    for (const c of completion) {
        participantMap.set(c, participantMap.get(c) - 1);
    }
    
    for (const [key, value] of participantMap) {
        if (value > 0) {
            return key;
        }
    }
}
